NetTopologySuite.IO.GeoPackage/GeoPackageGeoWriter.cs

```diff
@@ -0,0 +1,169 @@
+        { }
+
+        /// <summary>
+        /// Creates an instance of this class using the provided <see cref="ICoordinateSequenceFactory"/> and <see cref="IPrecisionModel"/> to use.
+        /// </summary>
+        public GeoPackageGeoWriter(ICoordinateSequenceFactory coordinateSequenceFactory, IPrecisionModel precisionModel)
+            : this(coordinateSequenceFactory, precisionModel, Ordinates.XYZM)
+        { }
+
+        /// <summary>
+        /// Creates an instance of this class using the provided <see cref="ICoordinateSequenceFactory"/> and <see cref="IPrecisionModel"/> to use.
+        /// Additionally the ordinate values that are to be handled can be set.
+        /// </summary>
+        public GeoPackageGeoWriter(ICoordinateSequenceFactory coordinateSequenceFactory, IPrecisionModel precisionModel, Ordinates handleOrdinates)
+        {
+            _coordinateSequenceFactory = coordinateSequenceFactory;
+            _precisionModel = precisionModel;
+            _handleOrdinates = handleOrdinates;
+        }
+
+        /// <inheritdoc cref="IGeometryWriter{TSink}.Write(IGeometry, Stream)"/>>
+        public void Write(IGeometry geom, Stream stream)
+        {
+            if (geom == null)
+            {
+                throw new ArgumentNullException(nameof(geom));
+            }
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+            using (var writer = new BinaryWriter(stream))
+            {
+                int byteOrder = (int)ByteOrder;
+                int ordinates;
+                switch (HandleOrdinates)
+                {
+                    case Ordinates.None:
+                        ordinates = 0;
+                        break;
+                    case Ordinates.XY:
+                        ordinates = 1;
+                        break;
+                    case Ordinates.XYZ:
+                        ordinates = 2;
+                        break;
+                    case Ordinates.XYM:
+                        ordinates = 3;
+                        break;
+                    case Ordinates.XYZM:
+                        ordinates = 4;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException("HandleOrdinates");
+                }
+                int isEmpty = geom.IsEmpty ? 1 : 0;
+                byte flags = (byte)(byteOrder + (ordinates << 1) + (isEmpty << 4));
+                var header = new GeoPackageBinaryHeader
+                {
+                    Extent = geom.EnvelopeInternal,
+                    Flags = flags
+                };
+                GeoPackageBinaryHeader.Write(writer, header);
+
+                bool emitZ = (HandleOrdinates & Ordinates.Z) == Ordinates.Z;
+                bool emitM = (HandleOrdinates & Ordinates.M) == Ordinates.M;
+                // NOTE: HandleSRID (always true) breaks wkb geopackage writes!
```

Comment:
:notebook: presumably, that's because SRID is set way earlier in the header.  SRID, to `WKBWriter`, is part of PostGIS's EWKB spec, which I guess is slightly different than the extensions that GeoPackage does here.

EWKB fits all its extension flags into the 5-byte WKB header (making it a superset of WKB), whereas just from reading this, GeoPackage seems to have decided that its blob format doesn't need to be a strict superset of WKB, which it uses to buy a more robust, more extensible format.

Reply:
so this code it's perfectly fine, I am right?

Reply:
the code should work, but maybe the comment could be updated to reflect the fundamental reason why we're doing things this way